Common++/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+add_library(
+  CommonPP
+  src/GeneralUtils.cpp
+  src/IpAddress.cpp
+  src/IpUtils.cpp
+  src/Logger.cpp
+  src/MacAddress.cpp
+  src/PcapPlusPlusVersion.cpp
+  src/SystemUtils.cpp
+  src/TablePrinter.cpp)
+
+set(public_headers
+    header/GeneralUtils.h
+    header/IpAddress.h
+    header/IpUtils.h
+    header/Logger.h
+    header/LRUList.h
+    header/MacAddress.h
+    header/PcapPlusPlusVersion.h
+    header/PointerVector.h
+    header/SystemUtils.h
+    header/TablePrinter.h
+    header/TimespecTimeval.h)
+
+set_property(TARGET CommonPP PROPERTY PUBLIC_HEADER ${public_headers})
+
+target_include_directories(
+  CommonPP
+  PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/header>
+         $<INSTALL_INTERFACE:include/pcapplusplus>
+  # Don't link with EndianPortable as it won't be exported
+  PRIVATE $<TARGET_PROPERTY:EndianPortable,INTERFACE_INCLUDE_DIRECTORIES>)
+
+if(TARGET_PLATFORM_WINDOWS)
+  target_link_libraries(CommonPP PUBLIC ws2_32 iphlpapi)
+endif()
+
+if(PCAPPP_INSTALL)
+  install(
+    TARGETS CommonPP
+    EXPORT PcapPlusPlusTargets
+    ARCHIVE DESTINATION ${PCAPP_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${PCAPP_INSTALL_LIBDIR}
+    PUBLIC_HEADER DESTINATION ${PCAPP_INSTALL_INCLUDEDIR})
+endif()
+
+set_property(TARGET CommonPP PROPERTY OUTPUT_NAME "common++")
+set_property(TARGET CommonPP PROPERTY VERSION ${VERSION_MAJOR}.${VERSION_MINOR})
```

Comment:
Where do `${VERSION_MAJOR}` and `${VERSION_MINOR}` come from? PcapPlusPlus uses date-based versioning scheme so "major" and "minor" have less significance, it's better to use one variable for version.

Same in [`CMakeLists.txt`](https://github.com/clementperon/PcapPlusPlus/blob/cmake-ng/Packet%2B%2B/CMakeLists.txt) and [`CMakeLists.txt`](https://github.com/clementperon/PcapPlusPlus/blob/cmake-ng/Pcap%2B%2B/CMakeLists.txt)

Reply:
Version is set based on git tag.

In case you in a dev branch or out of tree

version will be 22.05+g02b51784.20220731130355
And lib compile will get libPcap.so.22.05

Reply:
for official release
```
clement@Clement-Blade14:~/work/PcapPlusPlus/Dist$ git tag v23.01
clement@Clement-Blade14:~/work/PcapPlusPlus/Dist$ cmake ..
-- [MunkeiVersionFromGit] Version: 23.01
     Git tag:     [v23.01]
     Git branch:  [cmake-ng]
     Git hash:    []
     Decorated:   [v23.01]
     Identifiers: []
     Metadata:    []
     SemVer:      [23.01]
```